### PR TITLE
feat: arc mark

### PR DESCRIPTION
**Pull request by kanitw**
This may replace #5676 

- [x] modify all the specs to use theta/theta2

- [x] example: arc radial
- [x] example: layering text 
  - [x] add radius/theta for text 

- [x] support outerRadius / innerRadius in markDef and config -- pending   https://github.com/vega/vega/pull/2484

- [x] docs
  - [x] encoding.md
  - [x] arc.md 

- [x] radius(2)Offset

- [x] redirect theta/theta2/radius/radius2 config

- [x] test
  - [x] bin arc position
  - [x] radius scale range 

## LATER

- [ ] normalize angle to theta when reasonable https://github.com/vega/vega-lite/issues/6229

### CODE REVIEW

File: examples/compiled/layer_falkensee.vg.json [hunk 1]
```diff
@@ -126,7 +126,9 @@
       "encode": {
         "update": {
           "fill": {"scale": "color", "field": "event"},
-          "x": {"scale": "x", "field": "year_start"},
+          "x": {
```
Comment (kanitw) [inline examples/compiled/layer_falkensee.vg.json]: this is wrong

File: site/docs/encoding.md [hunk 2]
```diff
@@ -70,6 +70,7 @@ An integral part of the data visualization process is encoding data with visual
 The keys in the `encoding` object are encoding channels. Vega-Lite supports the following groups of encoding channels
 
 - [Position Channels](#position): `x`, `y`, `x2`, `y2`, `xError`, `yError`, `xError2`, `yError2`
+- TODO: add angular channels
```
Comment (kanitw) [inline site/docs/encoding.md]: Add docs

File: src/compile/mark/encode/valueref.ts [hunk 3]
```diff
@@ -323,3 +341,29 @@ export function widthHeightValueRef(channel: Channel, value: ValueOrGradientOrTe
   }
   return signalOrValueRef(value);
 }
+
+export function getValueFromMarkDefAndConfig({
```
Comment (kanitw) [inline src/compile/mark/encode/valueref.ts]: Hmm, this should be extracted. I'll extract and rebase later.

File: examples/compiled/layer_falkensee.vg.json (hunk 1 above)
Comment (domoritz) [reply examples/compiled/layer_falkensee.vg.json]: Is it still wrong?

Comment (kanitw) [reply examples/compiled/layer_falkensee.vg.json]: no diff anymore.

**Comment by domoritz** [comment]
Yet another example where not using the outline would be better :-P

Screen Shot 2020-03-31 at 21 07 11

File: examples/specs/arc_donut.vl.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
+  "description": "A simple donut chart with embedded data.",
+  "data": {
+    "values": [
+      {"id": 1, "field": 4},
```
Comment (domoritz) [inline examples/specs/arc_donut.vl.json:6]: id and field are weird names. Let's replace with `category` and `value`.

File: src/compile/mark/base.ts [hunk 5]
```diff
@@ -8,7 +9,7 @@ export interface MarkCompiler {
   /**
    * Underlying vega Mark type for the Vega-Lite mark.
```
Comment (domoritz) [inline src/compile/mark/base.ts:10]:
```suggestion
   * Underlying Vega Mark type for the Vega-Lite mark.
```

File: src/compile/mark/base.ts [hunk 6]
```diff
@@ -1,3 +1,4 @@
+import {Mark} from 'vega-typings/types';
```
Comment (domoritz) [inline src/compile/mark/base.ts:1]: Can we import from Vega instead?

File: test/compile/scale/range.test.ts [hunk 7]
```diff
@@ -264,6 +265,20 @@ describe('compile/scale', () => {
...
+      it('should use default radius.', () => {
+        const model = parseUnitModelWithScaleExceptRange({
+          mark: 'text',
+          encoding: {
+            radius: {field: 'x', type: 'quantitative'}
+          }
+        });
+        const r = parseRangeForChannel('radius', model);
+        expect(r.value[0]).toEqual(0);
+        expect(r.value[1]).toBeInstanceOf(SignalRefWrapper);
```
Comment (domoritz) [inline test/compile/scale/range.test.ts:278]:
```suggestion
        expect(r.value[1]).toEqual({signal: ...});
```

**Pull request merged**